include/seqan3/io/sam_file/output.hpp

```diff
@@ -534,11 +552,114 @@ class sam_file_output
      *
      * \include test/snippet/io/sam_file/push_back_tuple.cpp
      */
+#ifdef SEQAN3_DEPRECATED_310
+    template <typename tuple_t>
+    void push_back(tuple_t && t)
+    //!\cond
+        // new syntax enforces via static_assert that field::ref_seq, field::evalue, and field::bit_score isn't set
+        requires tuple_like<tuple_t> && (!detail::record_like<tuple_t>) && (!is_default_selected_field_ids)
+    //!\endcond
+    {
+        push_back_tuple(std::forward<tuple_t>(t));
+    }
+
+    // This is a bit of a problem, because to decide whether the new syntax or the old syntax is used is a bit
+    // complicated.
```

Comment:
```suggestion
    // The new syntax enforces via static_assert (see above) that field::ref_seq, field::evalue, and field::bit_score
    // isn't set. That makes sure that someone how explicitly requested for these fields will get a deprecation warning.
    // That leaves the case where the user did not provide any fields and just used the default ones. 
    // This causes a problem, because it is complicated to decide whether the new syntax or the old syntax is used.
```